middleware/middleware.go

```diff
@@ -0,0 +1,54 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-chi/chi"
+	"go.uber.org/zap"
+)
+
+// Middlewarer lets you add middlewares
```

Comment:
This feels like an odd naming convention.

Also, is there a way to specify the order of middleware layers or are we assuming they're idempotent?

Reply:
It's odd but it is idiomatic Go, e.g. see https://golang.org/doc/effective_go.html#interface-names